fdb-extensions/src/main/java/com/apple/foundationdb/async/hnsw/HNSW.java

```diff
@@ -0,0 +1,1945 @@
+            result = 31 * result + Boolean.hashCode(extendCandidates);
+            result = 31 * result + Boolean.hashCode(keepPrunedConnections);
+            result = 31 * result + Boolean.hashCode(useRaBitQ);
+            result = 31 * result + raBitQNumExBits;
+            return result;
+        }
+
+        @Override
+        @Nonnull
+        public String toString() {
+            return "Config[randomSeed=" + getRandomSeed() + ", metric=" + getMetric() +
+                    ", numDimensions=" + getNumDimensions() + ", isUseInlining=" + isUseInlining() + ", M=" + getM() +
+                    ", MMax=" + getMMax() + ", MMax0=" + getMMax0() + ", efConstruction=" + getEfConstruction() +
+                    ", isExtendCandidates=" + isExtendCandidates() +
+                    ", isKeepPrunedConnections=" + isKeepPrunedConnections() +
+                    ", useRaBitQ=" + isUseRaBitQ() +
+                    ", raBitQNumExBits=" + getRaBitQNumExBits() + "]";
+        }
+    }
+
+    /**
+     * Builder for {@link Config}.
+     *
+     * @see #newConfigBuilder
+     */
+    @CanIgnoreReturnValue
+    @SuppressWarnings("checkstyle:MemberName")
+    public static class ConfigBuilder {
+        private long randomSeed = DEFAULT_RANDOM_SEED;
+        @Nonnull
+        private Metric metric = DEFAULT_METRIC;
+        private boolean useInlining = DEFAULT_USE_INLINING;
+        private int m = DEFAULT_M;
+        private int mMax = DEFAULT_M_MAX;
+        private int mMax0 = DEFAULT_M_MAX_0;
+        private int efConstruction = DEFAULT_EF_CONSTRUCTION;
+        private boolean extendCandidates = DEFAULT_EXTEND_CANDIDATES;
+        private boolean keepPrunedConnections = DEFAULT_KEEP_PRUNED_CONNECTIONS;
+
+        private boolean useRaBitQ = DEFAULT_USE_RABITQ;
+        private int raBitQNumExBits = DEFAULT_RABITQ_NUM_EX_BITS;
+
+        public ConfigBuilder() {
+        }
+
+        public ConfigBuilder(final long randomSeed, @Nonnull final Metric metric, final boolean useInlining,
+                             final int m, final int mMax, final int mMax0, final int efConstruction,
+                             final boolean extendCandidates, final boolean keepPrunedConnections,
+                             final boolean useRaBitQ, final int raBitQNumExBits) {
+            this.randomSeed = randomSeed;
+            this.metric = metric;
+            this.useInlining = useInlining;
+            this.m = m;
+            this.mMax = mMax;
+            this.mMax0 = mMax0;
+            this.efConstruction = efConstruction;
+            this.extendCandidates = extendCandidates;
+            this.keepPrunedConnections = keepPrunedConnections;
+            this.useRaBitQ = useRaBitQ;
+            this.raBitQNumExBits = raBitQNumExBits;
+        }
+
+        public long getRandomSeed() {
+            return randomSeed;
+        }
+
+        @Nonnull
+        public ConfigBuilder setRandomSeed(final long randomSeed) {
```

Comment:
There should probably be a bit more documentation on what these values mean. I take it they are mainly constants from the HNSW paper, and that they've preserved the names used there, but it would still be good to document. Arguably, the Javadoc on the builder makes the most sense as that is presumably what the user will be looking at first (I guess)